app/Actions/Album/Tracks.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Actions\Album;
+
+use App\Models\Album;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Storage;
+
+class Tracks extends Action
+{
+	/**
+	 * @param string $albumID the ID of the album
+	 * @param mixed  $value   the value to be set
+	 */
+	public function set(string $albumID, UploadedFile $value): void
+	{
+		$album = $this->albumFactory->findAlbumOrFail($albumID, false);
+		if ($album->track_id != null) {
+			Storage::delete("tracks/$album->track_id.xml");
+		}
+
+		$new_track_id = uniqid();
+		Storage::putFileAs('tracks/', $value, "$new_track_id.xml");
+
+		if (Album::query()
+				->where('id', '=', $albumID)
+				->update(['track_id' => $new_track_id]) !== 1
```

Comment:
Please don't do that.

`UPDATE` queries are evil, because the don't update the `updated_at` attribute of the album properly. However, this will be necessary, when we will have other clients than the web frontend, because other clients might use caching of objects and need a proper `created_at` and `updated_at` value to perform a 3-way merge during syncronization.

Moreover, you already have the album object above, so you can use it. Do something like

```php
$album->track_short_path = $new_track_path;
$album->save()
```

Reply:
> `UPDATE` queries are evil, because the don't update the `updated_at`

On the other hand, such queries do not require the hydration of a model. :) 
I would be fine with update query, if they also include the `update_at` update (but that is nittpicking). :p